android/src/main/java/com/baseflow/googleapiavailability/GoogleApiAvailabilityManager.java

```diff
@@ -0,0 +1,147 @@
+
+import android.app.Activity;
+import android.app.Dialog;
+import android.app.PendingIntent;
+import android.content.Context;
+import android.util.Log;
+
+import com.google.android.gms.common.ConnectionResult;
+import com.google.android.gms.common.GoogleApiAvailability;
+
+import java.util.List;
+
+public class GoogleApiAvailabilityManager {
+
+    GoogleApiAvailability googleApiAvailability = GoogleApiAvailability.getInstance();
+
+    @FunctionalInterface
+    interface SuccessCallback {
+        void onSuccess(int connectionResult);
+    }
+
+    @FunctionalInterface
+    interface MakeGooglePlayServicesAvailableCallback {
+        void onSuccess(boolean makeGooglePlayServicesAvailable);
+    }
+
+    @FunctionalInterface
+    interface getErrorStringCallback {
+        void onSuccess(String errorString);
+    }
+
+    @FunctionalInterface
+    interface isUserResolvableCallback {
+        void onSuccess(boolean isUserResolvable);
+    }
+
+    @FunctionalInterface
+    interface showErrorNotificationCallback {
+        void onSuccess(boolean showErrorNotificationCallback);
+    }
+
+    @FunctionalInterface
+    interface showErrorDialogFragmentCallback {
+        void onSuccess(boolean showErrorDialogFragmentCallback);
+    }
+
+    @FunctionalInterface
+    interface ErrorCallback {
+        void onError(String errorCode, String errorDescription);
+    }
+
+    void checkPlayServicesAvailability(Boolean showDialog, Activity activity, Context applicationContext, SuccessCallback successCallback, ErrorCallback errorCallback) {
+        if (applicationContext == null || activity == null) {
+            Log.d(GoogleApiAvailabilityConstants.LOG_TAG, "Context and/or activity cannot be null.");
+            errorCallback.onError("GoogleApiAvailability.GoogleApiAvailabilityManager", "Android context and/or activity cannot be null.");
+            return;
+        }
+
+        final int connectionResult = googleApiAvailability
+                .isGooglePlayServicesAvailable(applicationContext);
+
+        if (showDialog != null && showDialog) {
+            googleApiAvailability
+                    .showErrorDialogFragment(activity, connectionResult, GoogleApiAvailabilityConstants.REQUEST_GOOGLE_PLAY_SERVICES);
+        }
+
+        successCallback.onSuccess(GoogleApiAvailabilityConstants.toPlayServiceAvailability(connectionResult));
+    }
+
+    void makeGooglePlayServicesAvailable(Activity activity, MakeGooglePlayServicesAvailableCallback successCallback, ErrorCallback errorCallback){
+        if (activity == null){
+            Log.d(GoogleApiAvailabilityConstants.LOG_TAG, "Activity cannot be null.");
+            errorCallback.onError("GoogleApiAvailability.makeGooglePlayServicesAvailable", "Android Activity cannot be null.");
+            return;
+        }
+
+        final boolean status = googleApiAvailability.makeGooglePlayServicesAvailable(activity).isSuccessful();
+        successCallback.onSuccess(status);
```

Comment:
Does `isSuccessful()` block and wait for the task to complete? I am not familiar with the `Task` API but I feel it would be better to add listeners to the returned task and use those instead:

```suggestion
        googleApiAvailability.makeGooglePlayServicesAvailable(activity)
            .addOnFailureListener((Exception e) -> successCallback.onSuccess(false))
            .addOnSuccessListener(() -> successCallback.onSuccess(true));
```